native/level1double.go

```diff
@@ -55,6 +58,9 @@ func (Implementation) Dnrm2(n int, x []float64, incX int) float64 {
 				sumSquares = sumSquares + (absxi/scale)*(absxi/scale)
 			}
 		}
+		if math.IsInf(scale, 1) {
+			return math.Inf(1)
```

Comment:
return scale rather than calling math.Inf(1)? Same for NaN and Inf elsewhere if so.